### DISCUSSION TITLE
Questions on CONFDIR and LOCAL_CONFDIR

### DISCUSSION THREAD

**Discussion by TonyGravagno** [rspamd/rspamd#3738] [Q&A] [answered]
When would the values of `CONFDIR` and `LOCAL_CONFDIR` be different?

In config files we see:
```
.include(...) "$LOCAL_CONFDIR/local.d/....inc"
.include(...) "$LOCAL_CONFDIR/override.d/....inc"
```
This [doc page](https://rspamd.com/doc/faq.html#rspamd-paths) agrees that the default values should be the same.
But isn't local.d called the "local configuration directory"?
And there are examples like `"${CONFDIR}/modules.d/*.conf"` which explicitly differentiates them.

How can we see the values of the environment variables that are actually being used by Rspamd at run-time?

And on a related topic seen above, is there any reason to prefer ${ENV_VAR} over $ENV_VAR in UCL? The code is inconsistent and there have been efforts to make the references consistent, like [this one](https://github.com/mailcow/mailcow-dockerized/commit/115c6540e2baddce97ec5c9b876f05f9250b1c7f). Is that something that will ever change in core?

Thanks!

**Comment by fatalbanana** [comment, marked as answer]
> When would the values of CONFDIR and LOCAL_CONFDIR be different?

Should LOCAL_CONFDIR be set to something else at build time.

> How can we see the values of the environment variables that are actually being used by Rspamd at run-time?

Run `rspamadm lua` and enter `rspamd_paths`

> And on a related topic seen above, is there any reason to prefer ${ENV_VAR} over $ENV_VAR in UCL? 

Probably no strong reason but being consistent couldn't hurt.

**Reply by TonyGravagno** [reply]
I'll get this info into the docs. TYVM
